platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/ConnectionServer.java

```diff
@@ -102,4 +109,26 @@ private void listen(final ServerSocket serverSocket) throws InterruptedException
             }
         }
     }
+
+    /**
+     * Adds a handshake completed listener
+     * */
+    private void addHandshakeListener(final Socket clientSocket) {
+        if (clientSocket instanceof final SSLSocket sslsocket) {
+            sslsocket.addHandshakeCompletedListener(event -> {
+                final PeerInfo peer = Utilities.validateTLSPeer(event.getSocket(), peerInfoList);
+                if (peer == null) {
+                    try {
+                        event.getSocket().close();
+                    } catch (final IOException e) {
+                        logger.warn(
```

Comment:
Should this be `error` instead of `warn`?

Reply:
same as the other, i think warn is ok.